rivulet2

```diff
@@ -97,9 +142,26 @@ parser.set_defaults(debug=False)
 args = parser.parse_args()
 starttime = time.time()
 img = loadimg(args.file)
-imgdtype = img.dtype
-img, crop_region = crop(img, args.threshold)  # Crop by default
 
+# Soma detection is run only when it is required Otherwise simple soma will be used
+if args.soma_detection:
+    smask = soma_detect(img, args.threshold, args.soma_smoothing, args.soma_lambda1, args.soma_lambda2, args.soma_iterations)
```

Comment:
Move the Soma class to soma.py. Embed your current methods in Soma class. 
Do soma_detect in r2(...)